homeassistant/components/plugwise/gateway.py

```diff
@@ -42,16 +46,17 @@ async def async_setup_entry_gw(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     try:
         connected = await api.connect()
-    except InvalidAuthentication:
-        LOGGER.error("Invalid username or Smile ID")
-        return False
-    except (ClientConnectionError, PlugwiseException) as err:
+    except ConnectionFailedError as err:
+        raise ConfigEntryNotReady("Failed to connect to the Plugwise Smile") from err
+    except InvalidAuthentication as err:
+        raise HomeAssistantError("Invalid username or Smile ID") from err
+    except (InvalidXMLError, ResponseError) as err:
         raise ConfigEntryNotReady(
-            f"Error while communicating to device {api.smile_name}"
+            "Error while communicating to the Plugwise Smile"
         ) from err
-    except asyncio.TimeoutError as err:
-        raise ConfigEntryNotReady(
-            f"Timeout while connecting to Smile {api.smile_name}"
+    except UnsupportedDeviceError as err:
+        raise HomeAssistantError(
+            "Unsupported device found: please create an Issue in the HA Core github"
```

Comment:
Same as above, this is not something a user should raise an issue for.

Reply:
Please see the answers above.

Reply:
Changed, as suggested.